core/src/main/java/com/datastrato/gravitino/utils/IsolatedClassLoader.java

```diff
@@ -214,11 +214,20 @@ private boolean isSharedClass(String name) {
         || name.startsWith("org.apache.log4j")
         || name.startsWith("org.apache.logging.log4j")
         || name.startsWith("java.")
-        || (name.startsWith("com.datastrato.gravitino.")
-            && !name.startsWith("com.datastrato.gravitino.catalog.hive."))
+        || !isCatalogClass(name)
         || sharedClasses.stream().anyMatch(name::startsWith);
   }
 
+  /**
+   * Checks if a given class name belongs to a catalog class.
+   *
+   * @param name The fully qualified class name.
+   * @return true if the class is a catalog class, false otherwise.
+   */
+  private boolean isCatalogClass(String name) {
+    return name.startsWith("com.datastrato.gravitino.catalog");
```

Comment:
Besides, I don't think condition `com.datastrato.gravitino.catalog` is enough to achieve our goal. please see 

<img width="1743" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/1eb8372d-2d91-4479-8dfc-06e91322ba3c">

Reply:
catalog module also relies on the core module, so it's not a problem?

Reply:
I will not have any problem currently, since you added the method `isSharedClass`, I think in this case, it will not reach our initial design as the class `core` module is indeed a shared class and does not need to be loaded by `IsolatedClassloader`

Reply:
Any conclusion on this? @mchades @yuqi1129 .

I think with this change the catalog package in `core` will be loaded in `IsolatedClassLoader`, not the base classloader?

Should we do like `! name.startsWith(xxx.hive or iceberg or jdbc or hadoop)`?

Reply:
> Any conclusion on this? @mchades @yuqi1129 .
> 
> I think with this change the catalog package in `core` will be loaded in `IsolatedClassLoader`, not the base classloader?
> 
> Should we do like `! name.startsWith(xxx.hive or iceberg or jdbc or hadoop)`?

+1

Reply:
updated